darts/dataprocessing/transformers/window_transformer.py

```diff
@@ -0,0 +1,691 @@
+import copy
+import itertools
+from typing import Iterator, List, Sequence, Tuple, Union
+
+import pandas as pd
+
+from darts.dataprocessing.transformers import BaseDataTransformer
+from darts.logging import get_logger, raise_if_not, raise_log, raise_user_warning
+from darts.timeseries import TimeSeries
+from darts.utils.utils import series2seq
+
+logger = get_logger(__name__)
+
+
+class ForecastingWindowTransformer(BaseDataTransformer):
```

Comment:
Two high level questions.
1) It seems somehow this doesn't work on multivariate series? I'd have expected to receive a multivariate series back with two windowed components when doing this:
![image](https://user-images.githubusercontent.com/4806678/195572071-279bfbeb-d024-4668-a04b-16eccc50c0e2.png)

2) What's the expected behavior when using several window functions? in the example below I see only one series being returned:
![image](https://user-images.githubusercontent.com/4806678/195572432-fafb869c-d821-47af-bf88-7a8998b06795.png)

Reply:
yes, true. The issue is that I have filtered components on which I apply the transformations and this loses the structure of the original series, because now they become only lists of univariate_series. I will revisit the filtering part and make sure that series structures are preserved. 
and for the multiple transformation case, there should be two resulting transformed series. I am not sure why they are not plotted both. Thank you for bringing that up.

Reply:
update : those issued should be solved now in TimeSeries.window_transform()

- if multivariate series, user can choose on which component to apply transformation and can choose if to return a TimeSeries with only the transformed components or the transformed components along the non transformed ones. The original components of the transformed ones are not returned.
- if series is stochastic, all samples are transformed identically, as per the specified corresponding component transformation 
-  if multiple transformations are desired, the result is a multivariate series with the different transformed data as new components. Similar behavior as previous for selecting and filtering components to apply transformation or not.